### ISSUE TITLE
pipeline config API: Cannot get pipeline config with full stop in the name

### ISSUE THREAD

**Issue by johncmckim** [gocd/gocd#1801]
I have created an API client to get config of a Go Pipeline. I have found that when the the pipeline name contains a `.` the api endpoint returns 404 with the following message:

```json
{
  "message": "Either the resource you requested was not found, or you are not authorized to perform this action."
}
```

I do have access to this pipeline normally so I suspect that this is a bug.

I would assume this is some kind of routing issue. I think that if the name component contains a dot e.g. `/api/admin/pipelines/someapp-3.0`, it's not being matched to method on the controller.

**Comment by ketan** [comment]
You may be right about the period on the pipeline name. Can you send the complete output of the curl command to eliminate any user error?

**Comment by johncmckim** [comment]
@ketan I've redacted some of the information but here is the output of two different pipelines.

![gocd-api-response](https://cloud.githubusercontent.com/assets/1297227/12316328/1a60efc2-bad1-11e5-98b8-7721e7560eb0.PNG)

**Comment by ketan** [comment]
This was expected to go in 16.1 but I submitted a PR #1802 from the wrong branch, the correct PR is #1915 which will be merged post 16.2

**Comment by rajiesh** [comment]
verified on 16.3.0-3178(b9741c2e994d09b65debe5d836aa428fd928e6db). period is now allowed in pipeline names
